[boost-exception] build failure

**Host Environment**
- OS: Ubuntu 20.04 LTS
- Compiler: revision

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install boost-exception:wasm32-emscripten`

**Failure logs** 
```
[1/2] cd /path/to/vcpkg/buildtrees/boost-exception/src/ost-1.78.0-8b4cb912e9.clean/build && /path/to/vcpkg/installed/x64-linux/tools/boost-build/b2 toolset=clang --user-config=/path/to/vcpkg/buildtrees/boost-exception/wasm32-emscripten-dbg/user-config.jam --stagedir=/path/to/vcpkg/buildtrees/boost-exception/wasm32-emscripten-dbg/stage --build-dir=/path/to/vcpkg/buildtrees/boost-exception/wasm32-emscripten-dbg runtime-link=shared link=static address-model=32 architecture=x86 target-os=linux threadapi=pthread variant=debug --layout=system -sICU_PATH="ICU_PATH-NOTFOUND" --with-atomic --with-random --with-date_time --with-filesystem --with-system --with-thread --with-chrono -j12 -sBOOST_ROOT=/path/to/vcpkg/installed/x64-linux/tools/boost-build -sBOOST_BUILD_PATH=/path/to/vcpkg/installed/x64-linux/tools/boost-build --debug-configuration --debug-building --debug-generators --ignore-site-config --hash -q debug-symbols=on -d +2 threading=multi stage
FAILED: CMakeFiles/boost /path/to/vcpkg/buildtrees/boost-exception/wasm32-emscripten-dbg/CMakeFiles/boost 
cd /path/to/vcpkg/buildtrees/boost-exception/src/ost-1.78.0-8b4cb912e9.clean/build && /path/to/vcpkg/installed/x64-linux/tools/boost-build/b2 toolset=clang --user-config=/path/to/vcpkg/buildtrees/boost-exception/wasm32-emscripten-dbg/user-config.jam --stagedir=/path/to/vcpkg/buildtrees/boost-exception/wasm32-emscripten-dbg/stage --build-dir=/path/to/vcpkg/buildtrees/boost-exception/wasm32-emscripten-dbg runtime-link=shared link=static address-model=32 architecture=x86 target-os=linux threadapi=pthread variant=debug --layout=system -sICU_PATH="ICU_PATH-NOTFOUND" --with-atomic --with-random --with-date_time --with-filesystem --with-system --with-thread --with-chrono -j12 -sBOOST_ROOT=/path/to/vcpkg/installed/x64-linux/tools/boost-build -sBOOST_BUILD_PATH=/path/to/vcpkg/installed/x64-linux/tools/boost-build --debug-configuration --debug-building --debug-generators --ignore-site-config --hash -q debug-symbols=on -d +2 threading=multi stage
notice: found boost-build.jam at /path/to/vcpkg/installed/x64-linux/tools/boost-build/boost-build.jam
notice: loading B2 from /path/to/vcpkg/installed/x64-linux/tools/boost-build/src/kernel/bootstrap.jam
notice: Site configuration files will be ignored due to the
notice: --ignore-site-config command-line option.
notice: Loading explicitly specified user configuration file:
    /path/to/vcpkg/buildtrees/boost-exception/wasm32-emscripten-dbg/user-config.jam
notice: Searching '/path/to/vcpkg/buildtrees/boost-exception/wasm32-emscripten-dbg' for user-config configuration file 'user-config.jam'.
notice: Loading user-config configuration file 'user-config.jam' from '/path/to/vcpkg/buildtrees/boost-exception/wasm32-emscripten-dbg'.
/path/to/vcpkg/installed/x64-linux/tools/boost-build/src/tools/common.jam:1171: in common.match-command-output from module common
error: '"/path/to/emsdk/upstream/emscripten/em++" --version' returned an invalid $kind string 'emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 3.1.1 (1934a98e709b57d3592b8272d3f1264a72c089e4)
Copyright (C) 2014 the Emscripten authors (see AUTHORS.txt)
This is free and open source software under the MIT license.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

'
/path/to/vcpkg/installed/x64-linux/tools/boost-build/src/tools/clang-linux.jam:101: in get-full-version from module clang-linux
/path/to/vcpkg/installed/x64-linux/tools/boost-build/src/tools/clang-linux.jam:106: in get-short-version from module clang-linux
/path/to/vcpkg/installed/x64-linux/tools/boost-build/src/tools/clang-linux.jam:55: in clang-linux.init from module clang-linux
/path/to/vcpkg/installed/x64-linux/tools/boost-build/src/build/toolset.jam:44: in toolset.using from module toolset
/path/to/vcpkg/installed/x64-linux/tools/boost-build/src/tools/clang.jam:33: in clang.init from module clang
/path/to/vcpkg/installed/x64-linux/tools/boost-build/src/build/toolset.jam:44: in toolset.using from module toolset
/path/to/vcpkg/installed/x64-linux/tools/boost-build/src/build/project.jam:1104: in using from module project-rules
/path/to/vcpkg/buildtrees/boost-exception/wasm32-emscripten-dbg/user-config.jam:11: in modules.load from module user-config
/path/to/vcpkg/installed/x64-linux/tools/boost-build/src/build-system.jam:255: in load-config from module build-system
/path/to/vcpkg/installed/x64-linux/tools/boost-build/src/build-system.jam:449: in load-configuration-files from module build-system
/path/to/vcpkg/installed/x64-linux/tools/boost-build/src/build-system.jam:607: in load from module build-system
/path/to/vcpkg/installed/x64-linux/tools/boost-build/src/kernel/modules.jam:294: in import from module modules
/path/to/vcpkg/installed/x64-linux/tools/boost-build/src/kernel/bootstrap.jam:139: in boost-build from module
/path/to/vcpkg/installed/x64-linux/tools/boost-build/boost-build.jam:8: in module scope from module

ninja: build stopped: subcommand failed.
```

**Additional context**
This package worked well until a few days ago. It seems like it fails to parse the compiler version.

--- a/ports/boost-build/0002-fix-get-version.patch
+++ b/ports/boost-build/0002-fix-get-version.patch
@@ -1,7 +1,7 @@
-diff --git a/tools/clang-linux.jam b/tools/clang-linux.jam
+diff --git a/src/tools/clang-linux.jam b/src/tools/clang-linux.jam
 index 949e277..8934787 100644
---- a/tools/clang-linux.jam
-+++ b/tools/clang-linux.jam
+--- a/src/tools/clang-linux.jam
++++ b/src/tools/clang-linux.jam
 @@ -97,7 +97,7 @@ rule init ( version ? :  command * : options * ) {
  
  rule get-full-version ( command-string )

--- a/ports/boost-build/0002-fix-get-version.patch
+++ b/ports/boost-build/0002-fix-get-version.patch
@@ -1,0 +1,13 @@
+diff --git a/tools/clang-linux.jam b/tools/clang-linux.jam
+index 949e277..8934787 100644
+--- a/tools/clang-linux.jam
++++ b/tools/clang-linux.jam
+@@ -97,7 +97,7 @@ rule init ( version ? :  command * : options * ) {
+ 
+ rule get-full-version ( command-string )
+ {
+-    return [ common.match-command-output version : "version ([0-9.]+)"
++    return [ common.match-command-output version : "([0-9]+.[0-9]+.[0-9]+)"
+            : "$(command-string) --version" ] ;
+ }
+ 

--- a/ports/boost-build/portfile.cmake
+++ b/ports/boost-build/portfile.cmake
@@ -12,7 +12,9 @@ vcpkg_from_github(
     REF boost-${BOOST_VERSION}
     SHA512 867966e3d254c0e996786587fb64ad1bda6f96546e5302c15231b17d66537798770bbd9e89f800d445a1f0a4d3be06dff8aed42dfd3a77b563d0f5d715e79324
     HEAD_REF master
-    PATCHES 0001-don-t-skip-install-targets.patch 0002-fix-get-version.patch
+    PATCHES
+        0001-don-t-skip-install-targets.patch
+        0002-fix-get-version.patch
 )
 
 vcpkg_download_distfile(ARCHIVE

--- a/ports/boost-build/portfile.cmake
+++ b/ports/boost-build/portfile.cmake
@@ -12,7 +12,7 @@ vcpkg_from_github(
     REF boost-${BOOST_VERSION}
     SHA512 867966e3d254c0e996786587fb64ad1bda6f96546e5302c15231b17d66537798770bbd9e89f800d445a1f0a4d3be06dff8aed42dfd3a77b563d0f5d715e79324
     HEAD_REF master
-    PATCHES 0001-don-t-skip-install-targets.patch
+    PATCHES 0001-don-t-skip-install-targets.patch 0002-fix-get-version.patch
 )
 
 vcpkg_download_distfile(ARCHIVE

--- a/ports/boost-build/vcpkg.json
+++ b/ports/boost-build/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "boost-build",
   "version": "1.78.0",
+  "port-version": 1,
   "description": "Boost.Build",
   "homepage": "https://github.com/boostorg/build",
   "dependencies": [

--- a/ports/boost-build/vcpkg.json
+++ b/ports/boost-build/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "boost-build",
   "version": "1.78.0",
-  "port-version": 1,
+  "port-version": 2,
   "description": "Boost.Build",
   "homepage": "https://github.com/boostorg/build",
   "dependencies": [

--- a/ports/boost-build/vcpkg.json
+++ b/ports/boost-build/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "boost-build",
   "version": "1.78.0",
-  "port-version": 2,
+  "port-version": 3,
   "description": "Boost.Build",
   "homepage": "https://github.com/boostorg/build",
   "dependencies": [

--- a/ports/boost-build/vcpkg.json
+++ b/ports/boost-build/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "boost-build",
   "version": "1.78.0",
-  "port-version": 3,
+  "port-version": 1,
   "description": "Boost.Build",
   "homepage": "https://github.com/boostorg/build",
   "dependencies": [

--- a/versions/b-/boost-build.json
+++ b/versions/b-/boost-build.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "a97c74b446de1829ea4b11d0b0b99f7081267c4d",
+      "git-tree": "cf970c17a4bf6d59deff7ce7e7eb98cec74b544a",
       "version": "1.78.0",
       "port-version": 1
     },

--- a/versions/b-/boost-build.json
+++ b/versions/b-/boost-build.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "b056cc4ec512fda610c9a46ed074136f87fe5784",
+      "version": "1.78.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "2c0d3c35e95f99911a226c3e736c0988f1139e51",
       "version": "1.78.0",
       "port-version": 0

--- a/versions/b-/boost-build.json
+++ b/versions/b-/boost-build.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "caada466764a8c06ee4e856b8a1c32745e1b5174",
+      "version": "1.78.0",
+      "port-version": 2
+    },
+    {
       "git-tree": "b056cc4ec512fda610c9a46ed074136f87fe5784",
       "version": "1.78.0",
       "port-version": 1

--- a/versions/b-/boost-build.json
+++ b/versions/b-/boost-build.json
@@ -3,16 +3,6 @@
     {
       "git-tree": "a97c74b446de1829ea4b11d0b0b99f7081267c4d",
       "version": "1.78.0",
-      "port-version": 3
-    },
-    {
-      "git-tree": "caada466764a8c06ee4e856b8a1c32745e1b5174",
-      "version": "1.78.0",
-      "port-version": 2
-    },
-    {
-      "git-tree": "b056cc4ec512fda610c9a46ed074136f87fe5784",
-      "version": "1.78.0",
       "port-version": 1
     },
     {

--- a/versions/b-/boost-build.json
+++ b/versions/b-/boost-build.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "a97c74b446de1829ea4b11d0b0b99f7081267c4d",
+      "version": "1.78.0",
+      "port-version": 3
+    },
+    {
       "git-tree": "caada466764a8c06ee4e856b8a1c32745e1b5174",
       "version": "1.78.0",
       "port-version": 2

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -574,7 +574,7 @@
     },
     "boost-build": {
       "baseline": "1.78.0",
-      "port-version": 0
+      "port-version": 1
     },
     "boost-callable-traits": {
       "baseline": "1.78.0",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -574,7 +574,7 @@
     },
     "boost-build": {
       "baseline": "1.78.0",
-      "port-version": 3
+      "port-version": 1
     },
     "boost-callable-traits": {
       "baseline": "1.78.0",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -574,7 +574,7 @@
     },
     "boost-build": {
       "baseline": "1.78.0",
-      "port-version": 1
+      "port-version": 2
     },
     "boost-callable-traits": {
       "baseline": "1.78.0",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -574,7 +574,7 @@
     },
     "boost-build": {
       "baseline": "1.78.0",
-      "port-version": 2
+      "port-version": 3
     },
     "boost-callable-traits": {
       "baseline": "1.78.0",

